ecosystem/indexer/src/indexer/fetcher.rs

```diff
@@ -28,15 +28,26 @@ impl TransactionFetcher {
             transactions_buffer: Default::default(),
         }
     }
+}
 
-    pub fn set_version(&mut self, version: u64) {
+#[async_trait::async_trait]
+impl TransactionFetcherTrait for TransactionFetcher {
+    fn set_version(&mut self, version: u64) {
         self.version = version;
     }
 
+    async fn fetch_ledger_info(&mut self) -> State {
+        self.client
+            .get_ledger_information()
+            .await
+            .expect("ledger info must be present")
```

Comment:
I don't suppose we can make this function return a result instead? We already panic all over the place in our codebase, it might be nice to start cleaning that up.

Reply:
I was trying to be consistent with the function below. But yea I guess we can.

Reply:
Actually I got an error. 
```
change the output type to match the trait: `Pin<Box<(dyn futures::Future<Output = aptos_rest_client::State> + std::marker::Send + 'async_trait)>>`
```